File: docs/src/testers/functions/invocations/check_run_status.ts
// @sniptest filename=check_run_status.ts
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();
const fn = client.NotteFunction({
  function_id: process.env.NOTTE_FUNCTION_ID!,
});

const result = await fn.run({ url: 'https://example.com' });
// Retrieve a run owned by this example, rather than a placeholder ID.
const runStatus = await fn.getRun(result.function_run_id);
console.log(
  JSON.stringify({
    status: runStatus.status,
    same_run: runStatus.function_run_id === result.function_run_id,
  }),
);

File: docs/src/testers/functions/invocations/check_run_status.py
# @sniptest filename=check_run_status.py
# @sniptest show=8-13
from notte_sdk import NotteClient

client = NotteClient()
function = client.Function("function_abc123")
run_id = "run_xyz789"

# Check run status
run_status = function.get_run(run_id)

print(f"Status: {run_status.status}")  # "active", "closed", "failed"
print(f"Result: {run_status.result}")
